src/ADNLPProblems/arglina.jl

```diff
@@ -1,12 +1,14 @@
 export arglina
 
 function arglina(; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
-  function f(x)
+  function F(x)
     n = length(x)
     m = 2 * n
-    return sum((x[i] - T(2 / m) * sum(x[j] for j = 1:n) - 1)^2 for i = 1:n) +
-           sum((-T(2 / m) * sum(x[j] for j = 1:n) - 1)^2 for i = (n + 1):m)
+    return vcat(
+      [x[i] - T(2 / m) * sum(x[j] for j = 1:n) - 1 for i = 1:n],
+      [-T(2 / m) * sum(x[j] for j = 1:n) - 1 for i = (n + 1):m]
+    )
   end
   x0 = ones(T, n)
-  return ADNLPModels.ADNLPModel(f, x0, name = "arglina"; kwargs...)
+  return ADNLPModels.ADNLSModel(F, x0, 2 * n, name = "arglina"; kwargs...)
```

Comment:
Yes, we could replace the definition `F` by the following:
```
function F(r, x)
    m = 2 * n
    for i=1:n
      r[i] = x[i] - T(2 / m) * sum(x[j] for j = 1:n) - 1
      r[i + n] = -T(2 / m) * sum(x[j] for j = 1:n) - 1
    end
    return r
  end
  function F(x)
    r = similar(x, 2 * n)
    return F(r, x)
  end
```
So that ADNLPModels will be (later) able to use the in-place function.

Reply:
Yes but let's call it `F!(r, x)`.

Reply:
There is a benefit of having both with the same name. If ADNLPModels can use the in-place function, then we wouldn't have to modify this, while if it's called `F!` we would have to modify this again for `ADNLSModel(F!, x0, 2 * n, name = "arglina"; kwargs...)`